File: java/lib/src/main/java/com/svix/api/Environment.java
// this file is @generated
package com.svix.api;

import com.svix.SvixHttpClient;
import com.svix.exceptions.ApiException;
import com.svix.models.EnvironmentIn;
import com.svix.models.EnvironmentOut;

import okhttp3.Headers;
import okhttp3.HttpUrl;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class Environment {
    private final SvixHttpClient client;

    public Environment(SvixHttpClient client) {
        this.client = client;
    }

    /** Download a JSON file containing all org-settings and event types. */
    public EnvironmentOut export() throws IOException, ApiException {

        return this.export(new EnvironmentExportOptions());
    }

    /** Download a JSON file containing all org-settings and event types. */
    public EnvironmentOut export(final EnvironmentExportOptions options)
            throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/environment/export");
        Map<String, String> headers = new HashMap<>();
        if (options.idempotencyKey != null) {
            headers.put("idempotency-key", options.idempotencyKey);
        }
        return this.client.executeRequest(
                "POST", url.build(), Headers.of(headers), null, EnvironmentOut.class);
    }

    /**
     * Import a configuration into the active organization.
     *
     * <p>It doesn't delete anything, only adds / updates what was passed to it.
     */
    public void import_(final EnvironmentIn environmentIn) throws IOException, ApiException {
        this.import_(environmentIn, new EnvironmentImportOptions());
    }

    /**
     * Import a configuration into the active organization.
     *
     * <p>It doesn't delete anything, only adds / updates what was passed to it.
     */
    public void import_(final EnvironmentIn environmentIn, final EnvironmentImportOptions options)
            throws IOException, ApiException {
        HttpUrl.Builder url = this.client.newUrlBuilder().encodedPath("/api/v1/environment/import");
        Map<String, String> headers = new HashMap<>();
        if (options.idempotencyKey != null) {
            headers.put("idempotency-key", options.idempotencyKey);
        }
        this.client.executeRequest("POST", url.build(), Headers.of(headers), environmentIn, null);
    }
}

File: kotlin/lib/src/main/kotlin/Environment.kt
// this file is @generated
package com.svix.kotlin

import com.svix.kotlin.models.EnvironmentIn
import com.svix.kotlin.models.EnvironmentOut
import okhttp3.Headers

data class EnvironmentExportOptions(val idempotencyKey: String? = null)

data class EnvironmentImportOptions(val idempotencyKey: String? = null)

class Environment(private val client: SvixHttpClient) {
    /**
     * Download a JSON file containing all org-settings and event types.
     *
     * Note that the schema for [`EnvironmentOut`] is subject to change. The fields herein are
     * provided for convenience but should be treated as JSON blobs.
     */
    suspend fun export(
        options: EnvironmentExportOptions = EnvironmentExportOptions()
    ): EnvironmentOut {
        val url = client.newUrlBuilder().encodedPath("/api/v1/environment/export")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }
        return client.executeRequest<Any, EnvironmentOut>(
            "POST",
            url.build(),
            headers = headers.build(),
        )
    }

    /**
     * Import a configuration into the active organization.
     *
     * It doesn't delete anything, only adds / updates what was passed to it.
     *
     * Note that the schema for [`EnvironmentIn`] is subject to change. The fields herein are
     * provided for convenience but should be treated as JSON blobs.
     */
    suspend fun import(
        environmentIn: EnvironmentIn,
        options: EnvironmentImportOptions = EnvironmentImportOptions(),
    ) {
        val url = client.newUrlBuilder().encodedPath("/api/v1/environment/import")
        val headers = Headers.Builder()
        options.idempotencyKey?.let { headers.add("idempotency-key", it) }

        client.executeRequest<EnvironmentIn, Boolean>(
            "POST",
            url.build(),
            headers = headers.build(),
            reqBody = environmentIn,
        )
    }
}
